src/fireedge/src/modules/features/General/slice.js

```diff
@@ -0,0 +1,324 @@
+                  mergedPayload
+                )
+              } else {
+                // Create new item
+                arrayAtPath[arrayAtPath.length] = mergedPayload
+              }
+            }
+
+            set(state.modifiedFields, pathWithoutIndex, arrayAtPath)
+          } else {
+            const nested = pathSegments?.length > 2
+            const nestedBasePath = nested
+              ? pathSegments.slice(0, -1).join('.')
+              : pathSegments.join('.')
+            const nestedKey = pathSegments.slice(-1)[0]
+
+            let existingNestedValue = get(
+              state.modifiedFields,
+              nestedBasePath,
+              {}
+            )
+
+            if (nested) {
+              existingNestedValue[nestedKey] = merge(
+                {},
+                existingNestedValue || {},
+                get(mergedPayload, pathSegments[0], {})
+              )
+            } else {
+              existingNestedValue = merge(
+                {},
+                existingNestedValue || {},
+                get(mergedPayload, pathSegments[0], {})
+              )
+            }
+
+            const parsedExisting = parsePayload(existingNestedValue, fieldPath)
+            set(state.modifiedFields, nestedBasePath, parsedExisting)
+          }
+        } else {
+          state.modifiedFields = merge({}, state.modifiedFields, mergedPayload)
+        }
+
+        if (payload?.setPath) {
+          state.fieldPath = payload.setPath
+        }
+      })
+      .addCase(actions.resetModifiedFields, (state) => {
+        state.modifiedFields = {}
+      })
+
+      /* UPLOAD NOTIFICATION */
+      .addCase(actions.setUploadSnackbar, (state, { payload }) => ({
+        ...state,
+        upload: payload,
+      }))
+      /* NOTIFICATION ACTIONS */
+      .addCase(actions.enqueueSnackbar, (state, { payload }) => {
+        const { key, options, message, values } = payload
+
+        return {
+          ...state,
+          notifications: [
+            ...state.notifications,
+            { key, options, message, values },
+          ],
+        }
+      })
+      .addCase(actions.dismissSnackbar, (state, { payload }) => {
+        const { key, dismissAll } = payload
+
+        return {
+          ...state,
+          notifications: state.notifications.map((notification) =>
+            dismissAll || notification.key !== key
+              ? { ...notification, dismissed: true }
+              : { ...notification }
+          ),
+        }
+      })
+      .addCase(actions.deleteSnackbar, (state, { payload }) => {
+        const { key } = payload
+
+        return {
+          ...state,
+          notifications: state.notifications.filter(
+            (notification) => notification.key !== key
+          ),
+        }
+      })
+
+      /*  REQUESTS API MATCHES */
+      .addMatcher(
+        ({ type }) => type.endsWith('/pending') && !type.includes('auth'),
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).